frontend/web/src/app/components/physical-print/physical-print-list/physical-print-list.component.ts

```diff
@@ -0,0 +1,163 @@
+      }, (error) => {
+        alert(error.verbose_message);
+      },
+    );
+  }
+
+  private populate() {
+    const _self = this;
+    /*manually generate columns*/
+    const columns = [ {
+      title: 'ID',
+      field: 'id',
+      sortable: true,
+    },{
+      title: 'Digital Print',
+      field: 'digitalPrintId',
+      sortable: true,
+    },{
+      title: 'SLM File',
+      field: 'slmPath',
+      sortable: true,
+    }, {
+      field: 'operate',
+      title: '<span class="glyphicon glyphicon-cog">',
+      align: 'center',
+      events: {
+        'click .edit': function(e, value, row, index) {
+          _self.router.navigate([_self.router.url, row.id]);
+        },
+        'click .delete': function(e, value, row, index) {
+          _self.delete(row.id);
+        },
+      },
+      formatter: this.operateFormatter,
+    }];
+
+    const data = [];
+    this.physicalPrints.forEach((physicalPrint) => {
+      data.push({
+        slmPath: physicalPrint.slmpath,
+        id: physicalPrint.id,
+        digitalPrintId : physicalPrint.digitalPrint
+      });
+    });
+
+    (this.table as any).bootstrapTable('destroy');
+    (this.table as any).bootstrapTable(
+      {
+        data,
+        columns,
+        sortStable: true,
+        sortName: 'name',
+      },
+    );
+  }
+
+  private prepareTriggers() {
+    const _self = this;
+    (this.table as any).on('click-row.bs.table', function(row, $element) {
+      _self.selected.emit(_self.physicalPrints.filter((physicalPrint) => {if (physicalPrint.id ===  $element.id)return physicalPrint; })[0]);
+    });
+  }
+
+  private amountFormatter(value, row, index) {
+    return value > 0 ? '' : {
+      css: {'background-color': 'rgba(255, 0, 0, 0.4)'},
+    };
+  }
+
+  private operateFormatter(value, row, index) {
+    return [
+      '<button class="edit btn btn-xs btn-primary" href="javascript:void(0)" title="Edit">',
+      '<i class="glyphicon glyphicon-pencil"></i>',
+      '</button>  ',
+      '<button  class="delete btn btn-xs btn-danger" href="javascript:void(0)" title="Delete">',
+      '<i class="glyphicon glyphicon-trash"></i>',
+      '</button>',
+    ].join('');
+  }
+
+  private delete(id) {
+    this.toBeDeleted = id;
+    this.openModal('#deleteFormDismissBtn');
+  }
+
+  private dismissDelete() {
+    this.toBeDeleted = null;
+    this.modalRef.hide();
+  }
+
+  private confirmDelete() {
+    if (this.toBeDeleted) {
+      this.physicalPrintService.deletePhysicalPrint(this.toBeDeleted).subscribe((res) => {
+          this.toBeDeleted = null;
+          this.modalRef.hide();
+          this.loadAndPopulate();
+        }, (error) => {
+          this.errorService.showAlert(error.verobose_message_header, error.verbose_message);
+        },
+      );
+    }
+  }
+
+  openModal(autoFocusIdWithHashtag: string) {
+    this.modalRef = this.modalService.show(this.modalDelete);
+    if (autoFocusIdWithHashtag != null && autoFocusIdWithHashtag != '') {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') != should be !== [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS1440)